kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -175,6 +182,64 @@ private MultipartUpload newMultipartUpload() throws IOException {
     }
   }
 
+  protected static class RetryWithExponentialBackoff {
```

Comment:
I commented above for the reasons on going for class implementation, but agree that in the current state of the code it makes sense to have it as a method. On the flip side handling retries is cross cutting concern which was also motivation to have it in a separate class and test it without unnecessary dependencies. Still believe that given number of retry unit tests covers quite a few use cases, but apparently I missed the one with sleep time and thus didn't catch XOR typo.